### ISSUE TITLE
Template class with the same name as existing class gets in the same output file

### ISSUE THREAD

**Issue by Xriuk** [doxygen/doxygen#8803]
**Describe the bug**
I have two classes: `MyClass` and `MyClass<P0>` the latter is an abstract template which gets implemented later on.
When I generate the documentation I get a single page for both classes which is simply called `...MyClass.html` and all my inheritance is messed up (classes inheriting from `MyClass` are not linked, ...).

**Expected behavior**
I should get two separate pages one for the class without parameters (`...MyClass.html`) and one for the template class (maybe you could count the number of parameters, in this case 1 becomes: `...MyClass_t1.html`, and if it had 2 parameters like `MyClass<P0, P1>` it could be `...MyClass_t2.html`)

**Screenshots**
![immagine](https://user-images.githubusercontent.com/6189813/135127625-13eaaae1-0347-46e6-87f7-6cf80d1a1044.png)
Here `MyClassChildTemplate` which inherits from `MyClass<bool>` is missing
![immagine](https://user-images.githubusercontent.com/6189813/135127790-b9d5d5e2-e926-4fed-ad1a-6341f4da5aba.png)
While here clicking on `MyClass<bool>` brings me to the `MyClass` page

**To Reproduce**
[Source.zip](https://github.com/doxygen/doxygen/files/7245952/Source.zip)

**Version**
1.9.2
Windows 10, 64 but

**Comment by Xriuk** [comment]
Forgot to mention that all the methods from the two classes are merged into one page

**Comment by sthames42** [comment]
I'm having this problem with a base class and two generics that inherit it. All show up on the same page.
Has there been any progress on this?

**Comment by Maxyeah** [comment]
We encountered the same issue after upgrading from version 1.8.8 to 1.12.0. 
This is a significant blocker for us, preventing the update!

**Comment by albert-github** [comment]
@maxyeah you wrote 1.8.8 is this correct (as this is from August 2014, so 10 years old ...)
Also:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by Maxyeah** [comment]
@albert-github Yes, that’s correct. Every time we attempt to update, we encounter issues that no one seems willing to address.

Now, I’m giving it another try, and hopefully, this will be the final issue.

It appears there have been multiple tickets related to this problem for years:

    [Issue 7281](https://github.com/doxygen/doxygen/issues/7281)
    [Issue 8803](https://github.com/doxygen/doxygen/issues/8803)

Each has uploaded an example. 

I can try to create another example, but seeing that there has been no progress for over four years makes me question whether it’s worth the effort.

**Comment by albert-github** [comment]
@Maxyeah 

When the mentioned examples cover your problem as well it is indeed not necessary to add an example.

I tried to run the example from this issue (#8803) with the 1.8.8 version and see indeed there is a difference, probably best shown on the page "annotated.html"

**1.8.8**
======
![image](https://github.com/user-attachments/assets/e2dce06b-e830-4064-a15d-704531d49c27)

**1.12.0**
======
![image](https://github.com/user-attachments/assets/400a36b0-e294-44d3-81fa-08be2c5e0639)


Note with 1.8.9 the `<P0>` disappeared so there was twice `MyClass` and with 1.9.0 theer was just one `MyClass`

**Comment by Maxyeah** [comment]
Yeah this looks like our issue.
**With 1.8.8**
<img width="687" alt="Screenshot_1_8_8" src="https://github.com/user-attachments/assets/ceb8e630-4d55-401e-bbd6-b1ab2e80d496">

**With 1.12.0**
<img width="628" alt="Screenshot_1_12_0" src="https://github.com/user-attachments/assets/499e148d-e549-4f9c-88ae-5d32ad9bacff">

I can share this example if necessary.

**Comment by albert-github** [comment]
When it is easy to share the example, please share as indicated in "Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent)."

**Comment by Maxyeah** [comment]
Here the Example.

[DoxyTest.zip](https://github.com/user-attachments/files/18027885/DoxyTest.zip)

**Comment by Maxyeah** [comment]
@albert-github Are there any updates on this issue?
Is this behavior expected, or is there an option to configure the output (e.g., multiple classes vs. a single class)?
We would greatly appreciate any updates or insights regarding this matter.

**Comment by albert-github** [comment]
> Are there any updates on this issue?

Not that I know of, I don't see quickly a possible change.

> Is this behavior expected, or is there an option to configure the output (e.g., multiple classes vs. a single class)?

I don't think this is expected and I don't know of a configuration setting for it.

**Comment by Maxyeah** [comment]
@doxygen @albert-github 
So we have to stay on this 

> so 10 years old ...

Version?

**Comment by Maxyeah** [comment]
Happy New Year @albert-github, @doxygen,

we saw you were very productive at the end of the year. Keep going!
We would appreciate some progress/feedback about this issue.

**Comment by doxygen** [comment]
@Maxyeah I'll look into it. Problem I see is that C# is the only languages where `A`, `A<T>` and `A<S,T>` are distinct classes, so Doxygen needs to do some form of name mangling to keep them in a single symbol map.

**Comment by Maxyeah** [comment]
Thank you for your efforts. We will take a look at it as soon as possible.

**Comment by Maxyeah** [comment]
@doxygen thanks for your efforts.
We tested the current master with the fix and it looks quite good. 
The classes are now separated in the tree, constructors and methods will be set correct.
Just the properties will be listed in the wrong class.
They are still all listed in the base Class.

![grafik](https://github.com/user-attachments/assets/6df9c6fd-a921-428e-8f45-ef0b7d1291d1)

[ExtendetDoxyTest.zip](https://github.com/user-attachments/files/18333065/ExtendetDoxyTest.zip)

**Comment by doxygen** [comment]
@Maxyeah Thanks for the example. There were indeed still some cases missing where names did not get properly mangled. Please check again with the latest commit.

**Comment by Maxyeah** [comment]
@doxygen works like a charm. Thanks, now we can finally update =D

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
